client-testkit/shared/src/main/scala/org/http4s/client/testkit/ClientRouteTestBattery.scala

```diff
@@ -205,4 +209,23 @@ abstract class ClientRouteTestBattery(name: String) extends Http4sSuite with Htt
     } yield true
   }
 
+  private def registerSuiteFixture[A](fixture: Fixture[A]): Fixture[A] = {
+    suiteFixtures += fixture
+    fixture
+  }
+
+  private def resourceSuiteDeferredFixture[A](
+      name: String,
+      resource: Resource[IO, A],
+  ): Fixture[IO[A]] =
+    registerSuiteFixture(UnsafeResourceSuiteLocalDeferredFixture(name, resource))
+
+  implicit private class ParseResultSyntax[A](self: ParseResult[A]) {
+    def yolo: A = self.valueOr(e => sys.error(e.toString))
+  }
+
+  private[this] lazy val suiteFixtures = List.newBuilder[Fixture[_]]
+
+  override def munitFixtures: Seq[Fixture[_]] = suiteFixtures.result()
```

Comment:
Had to inline these to drop the http4s testing dependency.

Reply:
I don't love this.  It seems like most of them are munit concerns, and either are missing from munit or us using munit inappropriately.  But I think this is sensible for now.

Reply:
Yep, agree. This weirdness gave me a heck of a time with with null pointer exceptions and initialization order issues.